include/rocksdb/iterator.h

```diff
@@ -117,6 +117,42 @@ class Iterator : public Cleanable {
     assert(false);
     return Slice();
   }
+
+  /**
+   * true if this iterator is already
+   * checking ReadOptions::iterate_lower_bound,
+   * false otherwise.
+   */
+  virtual bool check_lower_bound() const { return false; }
```

Comment:
`check_lower_bound` is not an existing or new stateful member, so why name it like an accessor with a verb name? How about `ChecksLowerBound` or `DoesCheckLowerBound`?